### ISSUE TITLE
Email subscription settings for individual bouts

### ISSUE THREAD

**Issue by carlosmiranda** [yegor256/netbout#547]
At the moment, any user/member of Netbout will receive an email for any message that arrives at any bout. It would be great if we could follow/subscribe to individual bouts. Emails should be sent only for new messages being posted into the subscribed ones.

**Comment by karato** [comment]
this task will get someone's attention soon

**Comment by karato** [comment]
@carlosmiranda many thanks for the report, I topped your account for 15 mins, transaction 50304140

**Comment by dmzaytsev** [comment]
@karato set milestone to 3.0

**Comment by karato** [comment]
@carlosmiranda milestone set to `3.1` (correct me if I am wrong)

**Comment by karato** [comment]
> @karato set milestone to 3.0

@dmzaytsev milestone set to `3.0`, thanks

**Comment by karato** [comment]
@ikhvostenkov do this task

**Comment by ikhvostenkov** [comment]
@carlosmiranda I've checked ```EmMessages``` and according with:
```Java

public void post(final String text) throws IOException {
        this.origin.post(text);
        for (final Friend friend : this.bout.friends().iterate()) {
            if (friend.email().isEmpty()) {
                continue;
            }
            this.email(friend, text);
        }
    }

```
Email messages will be delivered only for people who are added to current bout. I've checked it also with latest build and it works correctly. Could you, please, explain how were you able to reproduce this issue? Also you can unsubscribe from bout just by exiting this bout.

**Comment by carlosmiranda** [comment]
@ikhvostenkov Yes, that works fine, but we're talking here about a different scenario.  Let's say I still want to remain in the conversation, but I just want not to receive emails from it; I can't do that now. Users should be able to be participate in the bout but choose not to receive messages from them individually.

**Comment by yegor256** [comment]
@carlosmiranda @ikhvostenkov I think we will have to put a small icon, next to bout title, which can be triggered by a user, something like this: 

http://fortawesome.github.io/Font-Awesome/icon/bell-o/
http://fortawesome.github.io/Font-Awesome/icon/bell-slash-o/

**Comment by ikhvostenkov** [comment]
@karato Could you, please, assign someone else or postpone this task until 8th of June, I will be out until this for personal reasons?

**Comment by yegor256** [comment]
@karato give this to someone else pls

**Comment by karato** [comment]
> @karato give this to someone else pls

@yegor256 deducted 30 from your rating :(

**Comment by karato** [comment]
> @karato give this to someone else pls

@yegor256 no problem, I will try to find somebody else

**Comment by karato** [comment]
@erimerturk do this task

**Comment by erimerturk** [comment]
@carlosmiranda @yegor256 before this bug we need to fix another problem. To do this task i need to find friend's personal version of bout which will have subscription property. To access bout, find user, get inbox and find bout steps should be followed. At this version of user i am just able to find it by <b>URN</b>. Friend don't have <b>URN</b> property.  Friend has <code>alias</code> and <code>email</code> so i should be able to find user by one of them. I will open a bug for this task.

**Comment by erimerturk** [comment]
@carlosmiranda @yegor256 you can see PR details #672

**Comment by karato** [comment]
@carlosmiranda the code made here contains a puzzle `547-299902f6`/#682, which will be resolved soon

**Comment by erimerturk** [comment]
@carlosmiranda please check and close issue if it is acceptable, thank you

**Comment by carlosmiranda** [comment]
@erimerturk looks good, thanks!

**Comment by karato** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by elenavolokhova** [comment]
@erimerturk 
>karato commented on Jan 16
this task will get someone's attention soon

 According to our [quality rules](http://www.teamed.io/qa.html):
 > Messages in a ticket always start with a name of a user they are addressed to.
 
 As far as you are the performer of this ticket you are responsible for its quality. Please make sure that everyone follows this rule in your tickets in the future. Ask them to fix issues before closing the ticket.
 
 Please confirm that everything is clear and you have no questions regarding this approach.

**Comment by erimerturk** [comment]
@elenavolokhova i understand what do you say but i need to say that it happened before task assigned to me. Next time i will give more attention to it. thank you

**Comment by elenavolokhova** [comment]
@erimerturk It doesn't really matter when you were assigned to the task. You become responsible for ticket's quality as soon as you were assigned to it. Next time just ask to fix the issue, if you see it. Usually everyone is trying to help and fixes such minor things easily. Thanks!

**Comment by elenavolokhova** [comment]
@karato Quality is acceptable here.

**Comment by erimerturk** [comment]
@elenavolokhova next time i will try to fix it, thanks for comments

**Comment by karato** [comment]
> @karato Quality is acceptable here.

@elenavolokhova thanks a lot, next time everybody should try to make it better

**Comment by karato** [comment]
@erimerturk 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 59103324; I added **30 mins** to your account, many thanks for working with the project! The completion time here was AP-5PY025006B0016825.; +30 added to your rating, current score is: [+210](http://www.netbout.com/b/35280?open=rating)

**Comment by karato** [comment]
@carlosmiranda last puzzled solved `682-753e25af`/#693

**Comment by 0pdd** [comment]
@carlosmiranda all 3 puzzles are solved here: [#682](https://github.com/yegor256/netbout/issues/682), [#unknown](), [#unknown]().
